Code/Framework/AzCore/AzCore/Math/MathUtils.h

```diff
@@ -363,6 +363,13 @@ namespace AZ
         return a > b ? a : b;
     }
 
+    //! Returns a linear interpolation between 2 values.
+    template<typename T>
+    constexpr T Lerp(const T& a, const T& b, float t)
```

Comment:
not sure I really like this though when there is already an implementation under Vector?

Reply:
Hmm so the `Vector3` lerp is optimized to use SIMD operations, internally it looks like this...

```c++
AZ_MATH_INLINE Vector3 Vector3::Lerp(const Vector3& dest, float t) const
{
    return Vector3(Simd::Vec3::Madd(Simd::Vec3::Sub(dest.m_value, m_value), Simd::Vec3::Splat(t), m_value));
}
```

So keeping both might well make sense (you could always profile each of them to compare 😛)

Reply:
Also I think due to overload resolution rules this template will actually be used for all cases now, so the `float`/`double` implementations are technically redundant (might be worth checking up on this). I think someone from sig-core might be worth taking a look (thoughts @lumberyard-employee-dm?)

Reply:
we can get rid of the template and do something with constexpr to correctly handle those cases maaybe?

Reply:
I've added a few more reviewers to take a look, I think it does make sense to remove `MCore::LinearInterpolate` (if @amzn-jillich also agrees) but want to make sure others are okay with the new `Lerp` template. I'm fine with it in theory but I know in the recent past there has been discussion about avoiding templates for these sorts of things and I want to make sure other people get to weigh-in with any feedback/reasons why not to do it this way. If it's possible to just replace the usages of `MCore::LinearInterpolate` without adding the new template for now that might be the easiest option?

Reply:
Is this actually needed? You've ported the vec3/quat occurences to the `source.Lerp(target, weight)` versions already. Or are there many more?

Reply:
Awesome thanks @lumberyard-employee-dm, I think the template is then safe to add if we do want it. I remember there being something weird with templates around constructors being the 'best match' that could lead to some surprising behavior but good to know the overloads are selected first in this case 👍